lib/hooks/index.js

```diff
@@ -97,25 +67,51 @@ function findModuleVersion(request, parent, load) {
     if (mainScriptDir.indexOf(paths[i]) === 0) {
       // This is unsafe on windows if request contains a namespace
       // TODO: convert path separators in request
-      var pjson = load(path.join(paths[i], request, 'package.json'));
-      return pjson.version;
+      return path.join(paths[i], request);
     }
   }
   return null;
 }
 
+/**
+ * Determines the version of the module located at `modulePath`.
+ *
+ * @param {?string} modulePath The absolute path to the root directory of the
+ *    module being loaded. This may be null if we are loading an internal module
+ *    such as http.
+ */
+function findModuleVersion(modulePath, load) {
+  return modulePath ? load(path.join(modulePath, 'package.json')).version :
+    process.version;
+}
+
 function activate(agent) {
 
   logger = agent.logger;
 
-  function loadHookAndPatch(loadFn, instrumentation) {
-    assert(instrumentation.module);
-    assert(!instrumentation.hook);
+  function loadHookAndPatch(loadFn, instrumentation, moduleRoot, version) {
```

Comment:
1. Lets move this function and moduleAlreadyPatched inside the shimmer.wrap below. Get rid of the loadFn argument and reuse the originalModuleLoad from the closure.
2. It might be better to rename this to `loadAndPatch`.